cg/apps/lims/api.py

```diff
@@ -412,14 +418,34 @@ def get_sample_project(self, sample_id: str) -> str | None:
 
     def get_sample_rin(self, sample_id: str) -> float | None:
         """Return the sample RIN value."""
-        rin = None
+        rin: float | None = None
         try:
             sample_artifact: Artifact = Artifact(self, id=f"{sample_id}PA1")
             rin: float = sample_artifact.udf.get(PROP2UDF["rin"])
         except HTTPError as error:
             LOG.warning(f"Sample {sample_id} not found in LIMS: {error}")
         return rin
 
+    def get_sample_dv200(self, sample_id: str) -> float | None:
+        """Return the sample's percentage of RNA fragments greater than 200 nucleotides."""
+        dv200: float | None = None
+        try:
+            sample_artifact: Artifact = Artifact(self, id=f"{sample_id}PA1")
+            dv200: float = sample_artifact.udf.get(PROP2UDF["dv200"])
+        except HTTPError as error:
+            LOG.warning(f"Sample {sample_id} not found in LIMS: {error}")
+        return dv200
+
+    def get_sample_reception_qc_flag(self, sample_id: str) -> ReceptionQCFLag | None:
```

Comment:
This function will work for almost all preps, but we have some (like TGA for example) where the QC flag is set on an artifact further down in the dataflow. We have created a specific EPP in LIMS because of this. It saves the Initial QC outcome to the sample level on the UDF "Passed Initial QC" that has the potential string values "True" or "False". The function could therefore look something like this:

```python
def has_sample_passed_initial_qc(self, sample_id: str) -> bool | None:
    """Return boolean representing the outcome of the Initial QC protocol of the given sample."""
    boolean_outcome: bool | None = None
    try:
        sample: Sample = Sample(self, id=sample_id)
        boolean_outcome: bool = bool(sample.udf["Passed Initial QC"])
    except HTTPError as error:
        LOG.warning(f"Sample {sample_id} not found in LIMS: {error}")
    return boolean_outcome
``` 

Another option is to use `sample.udf.get(<udf name>)`, which will return `None` if the UDF value isn't present on the sample.

Reply:
Nice, thanks for the review and all the info. I've followed this suggestion 💯